fiftyone/core/utils.py

```diff
@@ -2065,6 +2095,357 @@ def get_alt_path(self, output_path, alt_dir=None):
+            )
+            + 1
+        )
+        while True:
+            # Add  file number to the output path with if necessary
+            if output_number > 1:
+                output_path = os.path.join(
+                    output_dir,
+                    output_name_sans_ext + f"-{output_number}" + output_ext,
+                )
+
+            try:
+                # Attempt to create a placeholder file to show  the output
+                # path has been claimed
+                touch_filename = os.path.basename(output_path)
+
+                if self.ignore_exts:
+                    touch_filename, _ = os.path.splitext(touch_filename)
+
+                touch_path = os.path.join(self.tmp_dir, touch_filename)
+                with open(touch_path, "x", encoding="utf-8"):
+                    ...
+
+            except FileExistsError:
+                # The output path has already been claimed with a placeholder
+
+                if self.idempotent:
+                    break
+            else:
+                # The output path was successfully claimed with a placeholder.
+                break
+
+            last_attempted_output_number = output_number
+
+            touch_prefix = os.path.join(self.tmp_dir, output_name_sans_ext)
+            touch_glob_pattern = glob.glob(glob.escape(touch_prefix) + "*")
+            touch_re_pattern = re.escape(touch_prefix) + r"-(\d+).*"
+
+            touch_paths = [
+                f
+                for f in touch_glob_pattern
+                if self.ignore_exts or os.path.splitext(f)[0] == output_ext
+            ]
+
+            touched_number = None
+            for touch_path in touch_paths:
+                if m := re.match(touch_re_pattern, touch_path):
+                    n = int(m.group(1))
+                    if touched_number is None or n > touched_number:
+                        touched_number = n
+
+            if touched_number is None:
+                touched_number = len(touch_paths)
+
+            touched_number += 1
+
+            output_number = (
+                touched_number
+                if touched_number > last_attempted_output_number
+                else last_attempted_output_number + 1
+            )
+
+        return output_path
+
+    def get_alt_path(self, output_path, alt_dir=None):
+        """Returns the alternate path for the given output path generated by
+        :meth:`get_output_path`.
+
+        Args:
+            output_path: an output path
+            alt_dir (None): a directory in which to return the alternate path.
+                If not provided, :attr:`alt_dir` is used
+
+        Returns:
+            the corresponding alternate path
+        """
+        root_dir = alt_dir or self.alt_dir or self.output_dir
+        rel_path = os.path.relpath(output_path, self.output_dir)
+        return os.path.join(root_dir, rel_path)
+
+
+class UniqueFilenameMaker(object):
+    """A class that generates unique output paths in a directory.
```

Comment:
We probably should add a note about parallel processing is only supported wihout chunk size

Reply:
It's already been added to the only place in the code that uses chunk size.